Sources_v3/Workers/Background/MessageSender.swift

```diff
@@ -111,7 +117,9 @@ private struct MessageSendingQueue<ExtraData: ExtraDataTypes> {
         let wasEmpty = self.requests.isEmpty
         requests.forEach { self.requests.insert($0) }
         if wasEmpty {
-            sendNextMessage()
+            dispatchQueue.async { [weak self] in
+                self?.sendNextMessage()
```

Comment:
Why don't we wrap `sendNextMessage` in `dispatchQueue.sync` so it's safe to call from everywhere?

Reply:
yep, that's a good point. The same was done in **MessageEditor** but with `perform`.
But I'd use `dispatchQueue.async` instead of `sync` not to block the calling thread

Reply:
fixed in f606817667ea1950595a786d1ff86ef4c5e4d197